### ISSUE TITLE
Exception being thrown from tartiflette

### ISSUE THREAD

**Issue by rjwills28** [dls-controls/coniql#34]
While running coniql locally I occasionally see exceptions being thrown by tartiflette. This occurred when testing the cs-web-proto and using coniql+cs-web-proto to monitor 100 PVs changing at a rate of 10Hz. 

Exception:
`Traceback (most recent call last):
  File "/home/cnuser/.local/share/virtualenvs/coniql-W5jWBl11/lib/python3.7/site-packages/tartiflette/execution/execute.py", line 200, in execute_operation
    fields,
  File "/home/cnuser/.local/share/virtualenvs/coniql-W5jWBl11/lib/python3.7/site-packages/tartiflette/execution/execute.py", line 147, in execute_fields
    return_exceptions=True,
concurrent.futures._base.CancelledError`

**Comment by rjwills28** [comment]
I have noticed that on occasion the Coniql application does not unsubscribe from all of the PVs when the websocket disconnects. For example when navigating away from a webpage displaying PV updates. 

I was debugging the subscribes and unsubscribes in Coniql and noticed this. If it does not unsubscribe from a PV, then when the page is reloaded, a new subscription is created on top of this and we start a memory leak as the first one never gets removed. 

I believe this issue is related to the exception detailed in this ticket. The `asyncio.gather` task [here](https://github.com/tartiflette/tartiflette/blob/421c1e937f553d6a5bf2f30154022c0d77053cfb/tartiflette/execution/execute.py#L161) gets cancelled, causing the exception and meaning that the rest of that method does not get executed. 

However something else must be happening underneath with the tasks that the `asyncio.gather` is running as I see this exception on occasions when Coniql does unsubscribe from all PVs and on times where it doesn't and starts a memory leak. 

A quick fix to avoid this exception is to surround the asyncio.gather with a try-except:
```
try:
    awaited = await asyncio.gather(*list(to_await.values()), return_exceptions=True)
    for index, result in zip(to_await, awaited):
        results[index] = result
except asyncio.CancelledError:
            pass
```
With this fix in place I found that I could no longer reproduce the scenario where Coniql does not unsubscribe from all PVs. The only problem of course is that this fix is in the tartiflette code.
